code/modules/mining/equipment/mineral_scanner.dm

```diff
@@ -55,6 +55,19 @@
 	range = 4
 	cooldown = 50
 
+/obj/item/t_scanner/adv_mining_scanner/toggle_on()
+	. = ..()
+	if(on)
+		START_PROCESSING(SSobj, src)
+	else
+		STOP_PROCESSING(SSobj, src)
+
+/obj/item/t_scanner/adv_mining_scanner/process()
+	if(!on)
+		STOP_PROCESSING(SSobj, src)
+		return null
+	scan()
+
```

Comment:
??

Reply:
wait im an idiot the base toggle_on() already did that

Reply:
wait the process() is copied too what was i doing